pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -5015,5 +5015,38 @@ public void removeAutoSubscriptionCreation(
                 });
     }
 
+    @GET
+    @Path("/{tenant}/{namespace}/{topic}/getMessageIdByIndex")
+    @ApiOperation(hidden = true, value = "Get Message ID by index.",
+            notes = "If the specified index is a system message, "
+                    + "it will return the message id of the later message.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 307, message = "Current broker doesn't serve the namespace of this topic"),
+            @ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Namespace or partitioned topic does not exist, "
+                    + "or the index is invalid"),
+            @ApiResponse(code = 406, message = "The topic is not a persistent topic"),
+            @ApiResponse(code = 412, message = "The broker is not enable broker entry metadata"),
+    })
+    public void getMessageIDByIndexAndPartitionID(@Suspended final AsyncResponse asyncResponse,
```

Comment:
The `AndPartitionID` suffix in the method seems meaningless (and confusing)?

Reply:
You are right! The suffix is incorrect.